builder/slide_and_fold_storage.cc

```diff
@@ -251,10 +251,10 @@ class slide_and_fold : public node_mutator {
     if (l.stage) {
       result = block::make({
           // Wait for the previous iteration of this stage to complete.
-          check::make(semaphore_wait(buffer_at(l.semaphores, std::vector<expr>{*l.stage, l.sym - l.step}))),
+          check::make(semaphore_wait(buffer_at(l.semaphores, std::vector<expr>{*l.stage, floor_div(l.sym - l.step, l.step)}))),
```

Comment:
I had thought about this when I was working on this, I was worried about the case where it wrapped around and wasn't aligned to the fold factor. However, thinking about it, it seems like l.sym should always be a multiple of `l.step`, so what you did here is correct, and then if it's correct, it seems like it should also simplify.

So in that case, should these expressions be written `floor_div(l.sym, l.step) - 1` instead?

Reply:
Hmm, if the loop min isn't a multiple of `l.step`, then this isn't true.

I guess what you have works because the remainder is always the same? That's a bit sketchy... should comment as such at least.

Reply:
I added a comment about it and changed the expression to one you suggested.